treeopt/treeOpt.py

```diff
@@ -0,0 +1,473 @@
+        :param problem: A function that takes takes parameters and returnes the
+        system responce
+        :type problem: Python function
+        :return: Nothing
+        :rtype: None
+
+        """
+        self.problem = problem
+
+    def set_problem_args(self, args):
+        """
+        Sets additional Arguments, which are to be passed with each call of the
+        problem function. Arguments have to be passed in the same order in
+        which the they are declared in the problem definition
+        :param args: Tuple containing the arguments
+        :type args: Tuple
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.problem_args = args
+
+    def set_sim_keyword(self, keyword):
+        """
+        Adds a keyword to the Optimization Algorithm. When a MEtamodell is to
+        be visualized, the value of the keyword is checked and depending on the
+        keyword a different plot is generated
+        :param keyword: Keyword describing the problem-type (default "extern",
+        other alternative "benchmark")
+        :type keyword: String
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.sim_keyword = keyword
+
+    def set_sm_method(self, method):
+        """
+        Sets the Method that is used to approximate the system responce in the
+        design space
+        :param method: one of the functions in modules/metamodell.py
+        :type method: python function
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.smMethod = method
+
+    def set_accuracy_criterion(self, method):
+        """
+        Sets the method that is used to define the termination condition of the
+        optimization workflow
+        :param method: one of the functions in modules/accuracy
+        :type method: Python Function
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.accuracyCriterion = method
+
+    def simulate_problem(self, x):
+        """
+        Executes the problem which is to be analyzed. If static Variables where
+        defined as additional arguments for the problem, these arguments are
+        added to the function call.
+        :param x: Point in the design space
+        :type x: Numpy-array
+        :return: System responce
+        :rtype: Numpy-array
+
+        """
+
+        if self.problem_args is None:
+            return self.problem(x)
+        else:
+            return self.problem(x, self.problem_args)
+
+    def start_optimization(self):
+        """
+        Function that starts the previosly parameterized adaptive optimization
+        loop
+        :return: Nothing
+        :rtype: None
+
+        """
+
+        self.x = self.samplingMethod(self.limits, self.numDOE)
+
+        self.y = self.simulate_problem(np.atleast_2d(self.x[0]))
+        for xi in self.x[1:]:
+            print(self.simulate_problem(np.atleast_2d(xi)))
```

Comment:
See above, `print` in a library.

Reply:
This was a Output for the V&V-Process. I deleted the line.